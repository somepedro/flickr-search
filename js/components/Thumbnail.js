import React from 'react'
import { object } from 'prop-types'
import { buildPhotoUrl } from '../helpers'

class Thumbnail extends React.Component {
  render () {
    const { photo } = this.props
    return (
      <div className='thumbnail-frame' key={photo.id}>
        <img className='thumbnail' src={buildPhotoUrl(photo)} title={photo.title} />
      </div>
    )
  }
}

Thumbnail.propTypes = {
  photo: object.isRequired
}

export default Thumbnail
