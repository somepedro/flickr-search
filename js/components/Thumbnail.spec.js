/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Thumbnail from './Thumbnail'
import samplePhotos from '../../data/sample_photos'

test('Thumbnail snapshot test', () => {
  const component = shallow(<Thumbnail photo={samplePhotos[0]} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
