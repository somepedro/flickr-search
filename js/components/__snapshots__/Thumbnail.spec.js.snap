// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail snapshot test 1`] = `
<div
  className="thumbnail-frame"
>
  <img
    className="thumbnail"
    src="https://farm5.staticflickr.com/4248/35335748176_d79bd5eea3.jpg"
    title="_MG_5475"
  />
</div>
`;
